engineyard/ey-cookbooks-stable-v5
# Updated Haproxy To Get Setting From Env-Vars

Description of your patch

Updated Haproxy to allow TLSv1.2 to be forcefully enabled disabling the other TLS settings this is done via a environmental variable so people can quickly enable and disable without editing chef, along with this change I've done the same for enabling HTTP2


Recommended Release Notes
N/A

Estimated risk
N/A

Components involved

- cookbooks/haproxy/attributes/haproxy.rb
- cookbooks/haproxy/libraries/helpers.rb
- cookbooks/haproxy/recipes/configure.rb
- cookbooks/haproxy/templates/default/haproxy.cfg.erb
- cookbooks/lb/libraries/helpers.rb
- cookbooks/lb/recipes/configure.rb
- cookbooks/lb/templates/default/haproxy.cfg.erb


Description of testing done

Check QA instructions - checked haproxy configuration to ensure certain configurations which are for TLSv1.2 (PCI compliance level) and HTTP2 are enabled

QA Instructions

- Create a test environment
- Overlay default haproxy and LB recipe with this one
- Upload and Apply chef.
- Enable a certificate on the environment
- Hit Apply to load certificate
- SSH into environment

**Check if default haproxy configuration works**

- Check haproxy configuration to ensure SSL certificate is loaded
- Extra check would be to curl the local address or domain using `curl -vvi https://127.0.0.1 -k`

**Check if TLSv1.2 only works**

- Go to Environment dashboard
- Click `environmental variables`
- Add variable called `EY_TLS_12` set value as `TRUE`
- Press apply on dashboard
- Check haproxy configuration to see if other cipher settings have been disabled should look like so:
` ssl-default-bind-options no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets`
- Extra check would be to curl the local address or domain using `curl -vvi https://127.0.0.1 -k --tlsv1.0` should generate an error

**Check if HTTP2 works**

- Go to Environment dashboard
- Click `environmental variables`
- Add variable called `EY_HTTP2` set value as `TRUE`
- Press apply on dashboard
- Check haproxy configuration to see if other cipher settings have been disabled should look like so:
` redirect scheme https code 301 if !{ ssl_fc }`


I will be updating the other appropriate files for HTTP2 configuration.
